pkg/clustermesh/clustermesh.go

```diff
@@ -240,3 +242,16 @@ func (cm *ClusterMesh) Status() (status *models.ClusterMeshStatus) {
 
 	return
 }
+
+func (cm *ClusterMesh) ExtendedClusterMeshEnabled() bool {
+	return cm.conf.ClusterInfo.MaxConnectedClusters != defaults.MaxConnectedClusters
+}
+
+// ValidateConnection validates a remote cluster config against runtime ClusterMesh options
+func (cm *ClusterMesh) ValidateConnection(cfg *cmtypes.CiliumClusterConfig) error {
+	if cm.ExtendedClusterMeshEnabled() && (cm.conf.ClusterInfo.MaxConnectedClusters != cfg.Capabilities.MaxConnectedClusters) {
+		return fmt.Errorf("mismatched MaxConnectedClusters; local=%d, remote=%d", cm.conf.ClusterInfo.MaxConnectedClusters, cfg.Capabilities.MaxConnectedClusters)
+	}
+
+	return nil
+}
```

Comment:
I think this (moving all into ClusterInfo.ValidateRemote or something similar) will require moving `ConfigValidationMode` to the ClusterInfo struct as well, otherwise it will be more difficult to test this aspect. Does this seems alright?

Reply:
Hmm, couldn't that be a parameter?

Reply:
updated with your feedback